spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/StringBasedAggregation.java

```diff
@@ -118,7 +119,17 @@ protected Object doExecute(MongoQueryMethod method, ResultProcessor resultProces
 
 			return result.getMappedResults();
 		}
-
+		
+		List mappedResults = result.getMappedResults(); 
+		
+		if(method.isPageQuery() || method.isSliceQuery()) {
```

Comment:
There's a slight difference between `Page` and `Slice`. Paging is aware of the total amount of items while slice requests `pageSize+1` to determine whether there's a next slice available.

Probably paging becomes more difficult because we have to issue a count query and that kind-of imposes performance questions when a complex aggregation is used.

For the time being, let's consider only `Slice` and I'm going to raise the issue with our team to discuss the performance aspect when someone wants to use  `Page<T>`.

Reply:
done the changes